File: raspi-js/ultrasoundDist.py
import RPi.GPIO as gpio
import time

TRIGER = 24
ECHO = 23

gpio.setmode(gpio.BCM)
gpio.setup(TRIGER, gpio.OUT)
gpio.setup(ECHO, gpio.IN)

startTime = time.time()

try:
	while True:
		gpio.output(TRIGER, gpio.LOW)
		time.sleep(0.1)
		gpio.output(TRIGER, gpio.HIGH)
		time.sleep(0.00002)
		gpio.output(TRIGER, gpio.LOW)

		while gpio.input(ECHO) == gpio.LOW:
			startTime = time.time()
		
		while gpio.input(ECHO) == gpio.HIGH:
			endTime = time.time()
		
		period = endTime - startTime
		dist1 = round(period * 1000000 / 58, 2)
		dist2 = round(period * 17241, 2)

		print("Dist1", dist1, "cm", ", Dist2", dist2, "cm")

		if dist1 < 10 and dist2 < 10:
			print("detect")
except:
	gpio.cleanup()

File: raspi-js/ultrasoundDist.js
const Gpio = require('pigpio').Gpio;
const sleep = require('util').promisify(setTimeout);

const TRIGER = 24;
const ECHO = 23;

const trig = new Gpio(TRIGER, { mode: Gpio.OUTPUT });
const echo = new Gpio(ECHO, { mode: Gpio.INPUT, alert: true });

trig.digitalWrite(0);

let startTime;
let endTime;

echo.on('alert', (level, tick) => {
  if (level === 1) {
    startTime = tick;
  } else if (level === 0) {
    endTime = tick;
    const period = endTime - startTime
    const dist1 = Math.round(period / 58 * 100) / 100;
    const dist2 = Math.round(period * 17241 / 1000000 * 100) / 100; 

    console.log("Dist1", dist1, "cm", ", Dist2", dist2, "cm");

    if (dist1 < 10 && dist2 < 10) {
      console.log("detect");
    }
  }
});

const measureDistance = () => {
  trig.trigger(10, 1); // Send a 10µs pulse
};

// Measure distance every 2 seconds
const interval = setInterval(measureDistance, 2000);

process.on('SIGINT', () => {
  clearInterval(interval); // Clear the interval on exit
  trig.digitalWrite(0);
  process.exit();
});
